### PR TITLE
Improve explain functionality

### PR DESCRIPTION
**Pull request by robwil**
Addresses the issues outlined in issue https://github.com/alcaeus/mongo-php-adapter/issues/188

### CODE REVIEW

File: lib/Alcaeus/MongoDbAdapter/AbstractCursor.php [hunk 1]
```diff
@@ -276,7 +276,7 @@ protected function getOptions($optionNames = null)
             $converter = 'convert' . ucfirst($option);
             $value = method_exists($this, $converter) ? $this->$converter() : $this->$option;
 
-            if ($value === null) {
+            if ($value === null || ($option === 'projection' && $value === [])) {
```
Comment (alcaeus) [inline lib/Alcaeus/MongoDbAdapter/AbstractCursor.php]: Line 277 checks to see if a `convert<Option>` method exists. This special case should be handled in a `convertProjection` method.

**Comment by alcaeus** [comment]
Thanks! Just a quick heads up, I'm heading out for a vacation, so I won't be able to merge these until I'm back in a couple of weeks. Sorry for the delay!

**Comment by robwil** [comment]
No worries!

**Comment by alcaeus** [comment]
Thanks @robwil!

**Comment by robwil** [comment]
Thanks for merging all of these @alcaeus . With these 4 fixes in place, all of our unit tests & integration tests passed on a very old legacy PHP application relying on the old mongo driver. Couldn't have done it so quickly without this adapter, so thanks for all your work on this!

**Comment by alcaeus** [comment]
Thank you for those kind words and your contributions, they are very much appreciated! 🍻

**Pull request merged**